### PR TITLE
Add icon to layer and fields comboboxes in "map value" widget's Load …

### PR DESCRIPTION
**Pull request by DelazJ**
…field from layer dialog

Before / After

image

### CODE REVIEW

**Comment by nyalldawson** [comment]
Hmm... any reason why we wouldn't want to use the standard QgsMapLayerComboBox/ QgsFieldCombo widgets here?

**Comment by DelazJ** [comment]
I simply forgot they exist. Thanks for the reminder. I will do the update.

**Comment by DelazJ** [comment]
I moved the widgets to use the dedicated field and map widgets and much better in the way the preview table is populated. And while at it, I added a help button and dialog geometry restoring.
I've played a lot with that dialog and i find the behavior quite confusing, but didn't address it here. This PR keeps the current behavior. See #59386 for the details on the issue.

File: src/gui/qgsattributetypeloaddialog.cpp [hunk 1]
```diff
@@ -120,9 +82,9 @@ void QgsAttributeTypeLoadDialog::createPreview( int fieldIndex, bool full )
     //when nothing is selected there is no reason for preview
     return;
   }
-  const int idx = keyComboBox->currentData().toInt();
-  const int idx2 = valueComboBox->currentData().toInt();
-  QgsMapLayer *dataLayer = QgsProject::instance()->mapLayer( layerComboBox->currentData().toString() );
+  const int idx = keyComboBox->currentIndex();
+  const int idx2 = valueComboBox->currentIndex();
+  QgsMapLayer *dataLayer = QgsProject::instance()->mapLayer( layerComboBox->currentLayer()->id() );
```
Comment (nyalldawson) [inline src/gui/qgsattributetypeloaddialog.cpp:87]:
```suggestion
  QgsMapLayer *dataLayer = layerComboBox->currentLayer();
```

File: src/gui/qgsattributetypeloaddialog.cpp [hunk 2]
```diff
@@ -169,9 +131,9 @@ bool QgsAttributeTypeLoadDialog::insertNull()
 void QgsAttributeTypeLoadDialog::loadDataToValueMap()
 {
   mValueMap.clear();
-  const int idx = keyComboBox->currentData().toInt();
-  const int idx2 = valueComboBox->currentData().toInt();
-  QgsMapLayer *dataLayer = QgsProject::instance()->mapLayer( layerComboBox->currentData().toString() );
+  const int idx = keyComboBox->currentIndex();
+  const int idx2 = valueComboBox->currentIndex();
+  QgsMapLayer *dataLayer = QgsProject::instance()->mapLayer( layerComboBox->currentLayer()->id() );
```
Comment (nyalldawson) [inline src/gui/qgsattributetypeloaddialog.cpp:136]:
```suggestion
  QgsMapLayer *dataLayer = layerComboBox->currentLayer();
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**